code/game/objects/items/devices/radio/headset.dm

```diff
@@ -111,6 +111,8 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 
 /obj/item/radio/headset/dropped(mob/user, silent)
 	. = ..()
+	if(user.get_item_by_slot(ITEM_SLOT_HEAD) != src || QDELETED(src)) //This can be called as a part of destroy
+		return
```

Comment:
What's with the slot flag check? This code (should be) safe regardless of it was equipped "properly" or not